src/osvr/Client/EyeTrackerRemoteFactory.cpp

```diff
@@ -0,0 +1,258 @@
+                                   util::time::TimeValue const &timestamp) {
+
+            OSVR_EyeTracker2DReport report;
+            report.sensor = data.sensor;
+            report.locationValid = false;
+            util::time::TimeValue timest = timestamp;
+
+            if (m_opts.reportLocation2D) {
+                report.locationValid =
+                    m_opts.locationIface->getState<OSVR_Location2DReport>(
+                        timest, report.state);
+            }
+            if (!(report.locationValid)) {
+                return; // don't send an empty report.
+            }
+
+            for (auto &iface : m_interfaces) {
+                iface->triggerCallbacks(timestamp, report);
+            }
+        }
+
+        void m_handleEyeBlink(common::OSVR_EyeNotification const &data,
+                              util::time::TimeValue const &timestamp) {
+
+            OSVR_EyeTrackerBlinkReport report;
+            report.sensor = data.sensor;
+            report.blinkValid = false;
+            util::time::TimeValue timest = timestamp;
+
+            if (m_opts.reportLocation2D) {
+                report.blinkValid =
+                    m_opts.locationIface->getState<OSVR_ButtonReport>(
+                        timest, report.state);
+            }
+            if (!(report.blinkValid)) {
+                return; // don't send an empty report.
+            }
+
+            for (auto &iface : m_interfaces) {
+                iface->triggerCallbacks(timestamp, report);
+            }
+        }
+
+        void m_handleEyeTracking(common::OSVR_EyeNotification const &data,
+                                 util::time::TimeValue const &timestamp) {
+            if (!m_all && *m_sensor != data.sensor) {
+                /// doesn't match our filter.
+                return;
+            }
+
+            m_handleEyeTracking3d(data, timestamp);
+            m_handleEyeTracking2d(data, timestamp);
+            m_handleEyeBlink(data, timestamp);
+        }
+
+        common::BaseDevicePtr m_dev;
+        common::InterfaceList &m_interfaces;
+        bool m_all;
+        Options m_opts;
+        boost::optional<OSVR_ChannelCount> m_sensor;
+    };
+
+    EyeTrackerRemoteFactory::EyeTrackerRemoteFactory(
+        VRPNConnectionCollection const &conns)
+        : m_conns(conns) {}
+
+    shared_ptr<RemoteHandler> EyeTrackerRemoteFactory::
+    operator()(common::OriginalSource const &source,
+               common::InterfaceList &ifaces, common::ClientContext &ctx) {
+
+        shared_ptr<RemoteHandler> ret;
+
+        NetworkEyeTrackerRemoteHandler::Options opts;
+
+        auto myDescriptor = source.getDeviceElement().getDescriptor();
+
+        if (myDescriptor["interfaces"]["eyetracker"].isMember("direction")) {
+            opts.reportDirection = true;
+            const std::string iface =
+                common::getPathSeparator() +
+                source.getDeviceElement().getDeviceName() + "/direction";
```

Comment:
So in this case this would have worked (adding a path sep to the beginning of the device name) but in general that can't be assumed - external VRPN devices break that assumption. Better to use `common::getFullPath` on the `common::PathNode`.